src/names/common.h

```diff
@@ -446,4 +448,33 @@ class CNameCache
 
 };
 
+/* ************************************************************************** */
+/* CQueuedTransaction.  */
+
+/**
+ * Keeps track of raw transactions that are queued for later broadcast.
+ * Example use cases:
+ *     Broadcast a name_firstupdate when its input name_new reaches maturity.
+ *     Broadcast a name_update when its input name_anyupdate nears expiration.
+ *     Broadcast a name_firstupdate when someone else's name expires.
+ * This data is serialized to the wallet so that the transaction can be
+ * broadcasted even if the client has restarted since the transaction was
+ * queued.
+ */
+class CQueuedTransaction
+{
+private:
+    CMutableTransaction tx;
```

Comment:
Out of curiosity, why is this using a `CMutableTransaction` and not `CTransaction`?  Do you have to modify it later on?

Reply:
> Have you tested that it works with descriptor wallets?

No, what's the easiest way to test this? Write some kind of integration test?

Reply:
Because CTransaction is more limited in terms of what APIs are available. You get a build error if you change it naïvely, so at the very least it would need more code.

Reply:
This does not work with descriptor wallets, but it does not work with descriptor wallets because `name_new` does not work with descriptor wallets. The reason it does not work is because `name_new` calls `getNameSalt` which calls `EnsureLegacyScriptPubKeyMan`. Oops!

This should be a new issue ("use `GetDescriptorScriptPubKeyMan` in `name_new`"), but logically shouldn't block merge of this because the code is already in the condition in which it is.

Reply:
To test with descriptor wallets, just run the test with `--descriptors` as argument.  And yes, `name_new` was broken until recently, but I fixed it in #432.  So it should be working now. :)

Reply:
Yes, this is fixed.